packages/twenty-server/src/modules/function/standard-objects/function.workspace-entity.ts

```diff
@@ -0,0 +1,100 @@
+import { Relation } from 'src/engine/workspace-manager/workspace-sync-metadata/interfaces/relation.interface';
+
+import { STANDARD_OBJECT_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-object-ids';
+import { WorkspaceEntity } from 'src/engine/twenty-orm/decorators/workspace-entity.decorator';
+import { WorkspaceIsSystem } from 'src/engine/twenty-orm/decorators/workspace-is-system.decorator';
+import { BaseWorkspaceEntity } from 'src/engine/twenty-orm/base.workspace-entity';
+import { WorkspaceField } from 'src/engine/twenty-orm/decorators/workspace-field.decorator';
+import { FUNCTION_STANDARD_FIELD_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids';
+import { FieldMetadataType } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { WorkspaceIsNullable } from 'src/engine/twenty-orm/decorators/workspace-is-nullable.decorator';
+import { WorkspaceRelation } from 'src/engine/twenty-orm/decorators/workspace-relation.decorator';
+import { RelationMetadataType } from 'src/engine/metadata-modules/relation-metadata/relation-metadata.entity';
+import { WorkspaceMemberWorkspaceEntity } from 'src/modules/workspace-member/standard-objects/workspace-member.workspace-entity';
+import { WorkspaceJoinColumn } from 'src/engine/twenty-orm/decorators/workspace-join-column.decorator';
+
+export enum FunctionSyncStatus {
+  NOT_READY = 'NOT_READY',
+  READY = 'READY',
+}
+
+@WorkspaceEntity({
+  standardId: STANDARD_OBJECT_IDS.function,
+  namePlural: 'functions',
+  labelSingular: 'Function',
+  labelPlural: 'Functions',
+  description: 'A function',
+})
+@WorkspaceIsSystem()
+export class FunctionWorkspaceEntity extends BaseWorkspaceEntity {
```

Comment:
As discussed briefly on Discord, I realized this should be a TypeORM object (it's kind of obvious here because you decide to put most of the logic in integration/core-module, but create this function object as a business modules). Functions are like ObjectMetadata, they are "meta" objects where we want more fine-grained control over permissions, etc.

Reply:
also added a comment above